### PR TITLE
Development 312 dashboard transfer

### PR DESCRIPTION
**Pull request by amiraabouhadid**
### Description

Added portal transfer page to playground 

### Changes

Added transfer by twin ID and transfer by account address. Used tfchain client for transfer function. 

### Related Issues

#1016 

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)
Screenshot from 2023-09-06 13-21-17
Screenshot from 2023-09-06 13-20-00

### CODE REVIEW

File: packages/playground/src/portal/transfer_view.vue [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  const twinAddress = receipientTwin.accountId;
+
+  const decimalAmount = new Decimal(transferAmount.value);
+  const milliAmount = decimalAmount.mul(10 ** 7).toNumber();
+  try {
+    const transferResult = await client.balances.transfer({
+      address: twinAddress,
+      amount: milliAmount,
+    });
+    await transferResult.apply();
```
Comment (AhmedHanafy725) [inline packages/playground/src/portal/transfer_view.vue:176]: we can use the `girdclient.balance.transfer` it already contains all this logic

File: packages/playground/src/portal/transfer_view.vue [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    createToast("Transaction submitted!", {
```
Comment (AhmedHanafy725) [inline packages/playground/src/portal/transfer_view.vue:177]: at this point, the transaction is already done, not submitted

File: packages/playground/src/portal/transfer_view.vue [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+}
+async function submitFormAddress() {
+  const twinId = await queryClient.twins.getTwinIdByAccountId({ accountId: receipientAddress.value });
+  const twinDetails = await queryClient.twins.get({ id: twinId });
+  if (twinDetails != null) {
+    loadingAddressTransfer.value = true;
+    await transfer(twinDetails);
+    loadingAddressTransfer.value = false;
+  } else {
+    createInvalidTransferToast("twin ID doesn't exist");
```
Comment (AhmedHanafy725) [inline packages/playground/src/portal/transfer_view.vue:197]: can't we validate the input and show the error in place instead of showing toast?

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: packages/playground/src/portal/transfer_view.vue [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+                </input-tooltip>
+              </input-validator>
+
+              <input-validator
+                :value="transferAmount"
+                :rules="[
+                  validators.required('Transfer amount is required '),
+                  validators.isNumeric('Amount should be a number.'),
+                  validators.min('Amount must be greater than 0', 0.00000000001),
+                  validators.max('Insuffient funds', freeBalance),
```
Comment (AhmedHanafy725) [inline packages/playground/src/portal/transfer_view.vue:35]:
```suggestion
                  validators.max('Insufficient funds', freeBalance),
```

**Comment by AhmedHanafy725** [comment]
Once I open the transfer page, it says that I don't have sufficient funds. however, I have trillions.
image

**Comment by AhmedHanafy725** [comment]
I can transfer to myself (my twinId 26) while this should be blocked.
image

**Comment by AhmedHanafy725** [comment]
twinId field should allow positive numeric values only
image

**Comment by AhmedHanafy725** [comment]
send button should be disabled when the input is invalid

image

**Comment by AhmedHanafy725** [comment]
clear button font color needs to be brighter like the send button as it looks disabled 

image

**Comment by AhmedHanafy725** [comment]
send button at least should be elevated and have a background color

image

**Comment by AhmedHanafy725** [comment]
give the selected tab color to be consistent with the rest components.

image

like this one

image

**Comment by AhmedHanafy725** [comment]
trim the white space on the address when validating
the address is valid, but it's invalid here as it has white space. the address might contain white space while coping it.
image

**Comment by AhmedHanafy725** [comment]
after transferring some tokens, it doesn't update the account balance as the current dashboard does.

**Comment by amiraabouhadid** [comment]
> after transferring some tokens, it doesn't update the account balance as the current dashboard does.

account balance is updated in profile_manager.vue which is outside of the scope of this pr.

File: packages/playground/src/portal/transfer_view.vue [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                </input-tooltip>
+              </input-validator>
+              <input-validator
+                v-if="!loadingBalance"
+                :value="transferAmount"
+                :rules="[
+                  validators.required('Transfer amount is required '),
+                  validators.isNumeric('Amount should be a number.'),
+                  validators.min('Amount must be greater than 0', 0.00000000001),
+                  validators.max('Insuffient funds', freeBalance),
```
Comment (AhmedHanafy725) [inline packages/playground/src/portal/transfer_view.vue:89]:
```suggestion
                  validators.max('Insufficient funds', freeBalance),
```

**Review by AhmedHanafy725 — COMMENTED** [review]
- the leading icon for transfer is missing

image

- `send` button background color seems different

image

image

File: packages/playground/src/portal/transfer_view.vue [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+const receptTwinFromTwinID = ref<Twin>();
+const gridClient = new GridClient({
+  mnemonic: useProfileManager().profile!.mnemonic,
+  network: window.env.NETWORK,
+});
+const freeBalance = ref(0);
+onMounted(async () => {
+  await gridClient.connect();
+  await getFreeBalance();
+});
```
Comment (MohamedElmdary) [inline packages/playground/src/portal/transfer_view.vue:148]: Please use `getGrid` util as it injects all urls which is important is custom networks (might be added later)

File: packages/playground/src/portal/transfer_view.vue (hunk 5 above)
Comment (amiraabouhadid) [reply packages/playground/src/portal/transfer_view.vue:89]: I really don't get why it shows up like this for you still been fixed in an earlier commit 
image

**Review by MohamedElmdary — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**